hoot-services/src/main/java/hoot/services/controllers/osm/user/UserResource.java

```diff
@@ -340,6 +344,154 @@ public Response getPrivileges(@Context HttpServletRequest request) {
+
+            String getFavorites = json.get("members").toString();
+
+            if ( !tags.containsKey(getName) ) {
+                tags.put(getName, getFavorites);
+                createQuery().update(users)
+                    .where(users.id.eq(userId))
+                    .set(users.favoriteOpts, tags)
+                    .execute();
+            }
+            else {
+                tags.remove(getName, getFavorites);
+                tags.put(getName, getFavorites);
+
+                createQuery().update(users)
+                        .where(users.id.eq(userId))
+                        .set(users.favoriteOpts, tags)
+                        .execute();
+            }
+
+
+        } catch (Exception e) {
+            Response.status(Status.INTERNAL_SERVER_ERROR).type(MediaType.TEXT_PLAIN).entity(e.getMessage()).build();
+        }
+       return Response.ok().build();
+   }
+
+    /**
+     * Gets the current users favorite adv opts
+     *
+     * GET hoot-services/osm/api/0.6/user/getFavoriteOpts
+     *
+     * @param request
+     * @return the current user's favorite adv opts
+     */
+    @GET
+    @Path("/getFavoriteOpts")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getFavoriteOpts(@Context HttpServletRequest request) {
+        Users user = Users.fromRequest(request);
+        Long userId = user.getId();
+
+        Object favoritesColumn = createQuery()
+                .select(users.favoriteOpts)
+                .from(users)
+                .where(users.id.eq(userId))
+                .fetchOne();
+
+        Map<String, String> json = PostgresUtils.postgresObjToHStore(favoritesColumn);
+
+        return Response.ok(json).build();
+    }
+
+    /**
+     * To delete saved favorite opts
+     *
+     * DELETE hoot-services/osm/api/0.6/user/deleteFavoriteOpts
+     * {
+     * 	"name":favOptName,
+     * }
+     *
+     * @param favoriteOpts favorite adv opt name
+     *            name of the opt to delete
+     * @return json containing total numbers of deleted
+     */
+    @DELETE
+    @Path("/deleteFavoriteOpts")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response deleteFavOpt(@Context HttpServletRequest request, String favoriteOpts) {
+
+        try {
+            Users user = Users.fromRequest(request);
+            Long userId = user.getId();
+
+            JSONParser parser = new JSONParser();
+            JSONObject json = (JSONObject) parser.parse(favoriteOpts);
+
+            String getName = (String) json.get("name");
+
+            Object favoritesColumn = createQuery()
+                    .select(users.favoriteOpts)
+                    .from(users)
+                    .where(users.id.eq(userId))
+                    .fetchOne();
+
+            Map<String, String> tags = PostgresUtils.postgresObjToHStore(favoritesColumn);
+
+            tags.remove(getName);
+
+            if ( !tags.containsKey(getName) ) {
```

Comment:
probably dont need to check if the key exists since the remove call above will have gotten rid of it